Source/Atomic/UI/UIWidget.h

```diff
@@ -203,6 +204,17 @@ class ATOMIC_API UIWidget : public Object, public tb::TBWidgetDelegate
 
     void DeleteAllChildren();
 
+    /// searches for specified widget ID from the top of the widget tree, returns the 1st one found.
+    virtual UIWidget *FindWidget ( const String& searchid );
+    /// return all of the widgets of the specified classname and id that is not 0 from the current widget
+    virtual void SearchWidgetClass ( const String& className, UISelectItemSource *results );  
+    ///  return all of the widgets of the specified id and id that is not 0 from the current widget
+    virtual void SearchWidgetId ( const String& searchid, UISelectItemSource *results );
+    /// return all of the widgets with the specified text and id that is not 0 from the current widget
+    virtual void SearchWidgetText ( const String& searchText, UISelectItemSource *results );
+    /// print out the widget tree to stdout from the current widget
```

Comment:
My rational was I needed a dynamic container that already existed. I though (briefly) about using std::vector as the container, but didn't want to add that dependency to TB, I tried the TB LinkedList, but that was a disaster. So that left UISelectItemSource, which works ok.  Even though it will ignore anonymous id widgets in the scripting languages, you're going to have to _grok_ Turbobadger pretty good to know what you're missing, and you still have the option of putting an id on a widget if you want to talk to it.